src/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -494,19 +525,17 @@ internal bool RemoveParsedValue(string name, object value)
                 // If there is no value for the header left, remove the header.
                 if (info.IsEmpty)
                 {
-                    bool headerRemoved = Remove(name);
-                    Debug.Assert(headerRemoved, "Existing header '" + name + "' couldn't be removed.");
+                    bool headerRemoved = Remove(descriptor);
+                    Debug.Assert(headerRemoved, "Existing header '" + descriptor.Name + "' couldn't be removed.");
```

Comment:
I agree this would be better -- but this is preexisting code so I don't think it needs to be addressed here.